addon/mixins/attachable.js

```diff
@@ -32,12 +32,12 @@ export default Ember.Mixin.create({
       if (!Ember.isNone(data[key])) {
         if (Ember.isArray(data[key])) {
           return data[key].forEach(function(val) {
-            return formData.append("" + root + "[" + key + "][]", val);
+            return formData.append("" + root + "[" + key + "][]", val, val.filename || 'blob');
```

Comment:
@esbanarango ah very true (it was a quick proof-of-concept, a bit rushed really)

Can you think of a way we can introspect those values, to see if they're an attachment or not? I.e. could we do some checks like `val instanceof Blob`?